packages/turbo-codemod/src/commands/migrate/steps/getTurboUpgradeCommand.ts

```diff
@@ -0,0 +1,182 @@
+  };
+}
+
+function getGlobalUpgradeCommand(
+  packageManager: PackageManager,
+  to: string = "latest"
+) {
+  switch (packageManager) {
+    case "yarn":
+      return `yarn global add turbo@${to}`;
+    case "npm":
+      return `npm install turbo@${to} --global`;
+    case "pnpm":
+      return `pnpm install turbo@${to} --global`;
+  }
+}
+
+function getLocalUpgradeCommand({
+  packageManager,
+  packageManagerVersion,
+  installType,
+  isUsingWorkspaces,
+  to = "latest",
+}: {
+  packageManager: PackageManager;
+  packageManagerVersion: string;
+  installType: InstallType;
+  isUsingWorkspaces?: boolean;
+  to?: string;
+}) {
+  const renderCommand = (
+    command: Array<string | boolean | undefined>
+  ): string => command.filter(Boolean).join(" ");
+  switch (packageManager) {
+    // yarn command differs depending on the version
+    case "yarn":
+      // yarn 2.x and 3.x (berry)
+      if (gte(packageManagerVersion, "2.0.0")) {
+        return renderCommand([
+          "yarn",
+          "add",
+          `turbo@${to}`,
+          installType === "devDependencies" && "--dev",
+        ]);
+        // yarn 1.x
+      } else {
+        return renderCommand([
+          "yarn",
+          "add",
+          `turbo@${to}`,
+          installType === "devDependencies" && "--dev",
+          isUsingWorkspaces && "-W",
+        ]);
+      }
+    case "npm":
+      return renderCommand([
+        "npm",
+        "install",
+        `turbo@${to}`,
+        installType === "devDependencies" && "--save-dev",
+      ]);
+    case "pnpm":
+      return renderCommand([
+        "pnpm",
+        "install",
+        `turbo@${to}`,
+        installType === "devDependencies" && "--save-dev",
+        isUsingWorkspaces && "-w",
+      ]);
+  }
+}
+
+function getInstallType({ directory }: { directory: string }): {
+  installType?: InstallType;
+  isUsingWorkspaces?: boolean;
+} {
+  // read package.json to make sure we have a reference to turbo
+  const packageJsonPath = path.join(directory, "package.json");
+  const pnpmWorkspaceConfig = path.join(directory, "pnpm-workspace.yaml");
+  const isPnpmWorkspaces = fs.existsSync(pnpmWorkspaceConfig);
+
+  if (!fs.existsSync(packageJsonPath)) {
+    console.error(`Unable to find package.json at ${packageJsonPath}`);
+    return { installType: undefined, isUsingWorkspaces: undefined };
+  }
+
+  const packageJson = fs.readJsonSync(packageJsonPath);
+  const isDevDependency =
+    packageJson.devDependencies && "turbo" in packageJson.devDependencies;
+  const isDependency =
+    packageJson.dependencies && "turbo" in packageJson.dependencies;
+  let isUsingWorkspaces = "workspaces" in packageJson || isPnpmWorkspaces;
+
+  if (isDependency || isDevDependency) {
+    return {
+      installType: isDependency ? "dependencies" : "devDependencies",
+      isUsingWorkspaces,
+    };
+  }
+
+  return {
+    installType: undefined,
+    isUsingWorkspaces,
+  };
+}
+
+/*
```

Comment:
Minor thing, but I believe JSDoc expects /** for doc comments which I believe is a convention most js docgen tools now expect also.

Reply:
Ah good call, will fix!